indico/modules/designer/pdf.py

```diff
@@ -92,22 +94,38 @@ def get_pdf(self):
         return data
 
     def _draw_item(self, canvas, item, tpl_data, content, margin_x, margin_y):
-        style = ParagraphStyle({})
-        style.alignment = ALIGNMENTS[item['text_align']]
-        style.textColor = item.get('color') or '#000000'
-        style.backColor = item.get('background_color') or None
-        style.borderPadding = (0, 0, 4, 0)
-        style.fontSize = _extract_font_size(item['font_size'])
-        style.leading = style.fontSize
+        styles = {
+            'alignment': ALIGNMENTS[item['text_align']],
+            'textColor': item.get('color') or '#000000',
+            'backColor': item.get('background_color') or None,
+            'borderPadding': (0, 0, 4, 0),
+            'fontSize': _extract_font_size(item['font_size'])
+        }
+        styles['leading'] = styles['fontSize']
 
         if item['bold'] and item['italic']:
-            style.fontName = FONT_STYLES[item['font_family']][3]
+            styles['fontName'] = FONT_STYLES[item['font_family']][3]
         elif item['italic']:
-            style.fontName = FONT_STYLES[item['font_family']][2]
+            styles['fontName'] = FONT_STYLES[item['font_family']][2]
         elif item['bold']:
-            style.fontName = FONT_STYLES[item['font_family']][1]
+            styles['fontName'] = FONT_STYLES[item['font_family']][1]
         else:
-            style.fontName = FONT_STYLES[item['font_family']][0]
+            styles['fontName'] = FONT_STYLES[item['font_family']][0]
+
+        for update in values_from_signal(
+            signals.event.designer.customize_badge_style.send(item, styles=styles),
+            as_list=True
+        ):
+            styles.update(update)
+
+        style = ParagraphStyle({})
```

Comment:
I didn't look more closely, but this `{}` feels like you might be able to pass `styles` directly instead of assigning one by one. Maybe you can check if that's indeed the case?

Reply:
It was not possible to pass the dict to the `ParagraphStyle` like this, the values were not added and the badge was printed without any style